### DISCUSSION TITLE
oracle cdc 2.1无法运行

### DISCUSSION THREAD

**Discussion by X1aoC** [apache/flink-cdc#698] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/41811521/145173073-0b29f706-61c6-42aa-89c4-dac0ada4667c.png)
![image](https://user-images.githubusercontent.com/41811521/145173116-aa658258-715a-4f7b-aa7b-3e0761bd6d60.png)
![image](https://user-images.githubusercontent.com/41811521/145173299-5c472f28-6fd5-443f-b8ce-15e4db46f0ed.png)
![image](https://user-images.githubusercontent.com/41811521/145173530-06c2b6df-3fb2-49a8-99df-0fc277c57bb8.png)
![image](https://user-images.githubusercontent.com/41811521/145173693-2bd42098-c919-43db-9e3b-1b88eee8fcb9.png)
![image](https://user-images.githubusercontent.com/41811521/145173737-3d21b2d2-f73d-4694-b52d-93b5781b0398.png)

按照官方文档操作，但最后报了这个错误，实在找不出原因，请求帮助

**Comment by GeiChan** [comment, marked as answer]
我今天也遇到了该问题，依赖问题导致的。我猜测，您使用的Driver也是 ojdbc6。更改为ojdbc8即可。
两个都存在时，有依赖冲突问题，排除掉ojdbc6即可
ojdbc6的getColumns获取的参数size只有18，而ojdbc8的getColumns获取的参数size为23.
debezium 的一段代码中有这样一段内容
```java
  ColumnEditor column = Column.editor().name(columnName);
  column.type(columnMetadata.getString(6));
  column.length(columnMetadata.getInt(7));
  if (columnMetadata.getObject(9) != null) {
      column.scale(columnMetadata.getInt(9));
  }

  column.optional(isNullable(columnMetadata.getInt(11)));
  column.position(columnMetadata.getInt(17));
  column.autoIncremented("YES".equalsIgnoreCase(columnMetadata.getString(23)));   // 主要问题所在，使用ojdbc6时，索引越界了
  ...
```
希望对您有所帮助。

**Reply by X1aoC** [reply]
非常感谢您的帮助
经过确认，我这边使用的是ojdbc8，更换了其他版本也是不行，非常奇怪
![image](https://user-images.githubusercontent.com/41811521/145372095-be1543d5-b2c7-478d-9e23-53642f982057.png)

**Comment by GeiChan** [comment]
<img width="1145" alt="oracle2kafkaException" src="https://user-images.githubusercontent.com/42415224/145385095-35174fe9-9199-48e7-ab94-2baa4d086af9.png">
可以尝试着在这个地方打个断点，计算下这个结果。应该就是这里的问题。
我是更改driver解决的。
暂时无法提供更多的帮助了，抱歉。

**Reply by X1aoC** [reply]
找到原因了，确实是驱动问题，用了默认路径下的驱动，更改后运行成功，非常感谢
